ops/terraform/services/common/scripts/destroy-autoscaled-nodes.sh

```diff
@@ -0,0 +1,111 @@
+trimmed_cluster_id="$(echo "$DB_CLUSTER_ID" | tr -d '[:space:]')"
+if [[ -z $trimmed_cluster_id ]]; then
+  echo "DB_CLUSTER_ID must not be an empty string or whitespace"
+  exit 1
+fi
+
+# jq expression returns the first element in the returned array as a precaution against a scenario
+# where multiple clusters are erroneously returned. Otherwise, the nodes in all clusters would be
+# marked for deletion. Additionally, we select for only the nodes with names containing
+# "autoscaling" to avoid possibly deleting the writer node which is managed by Terraform directly
+nodes_in_cluster="$(
+  aws rds describe-db-clusters --db-cluster-identifier "$trimmed_cluster_id" |
+    jq -r '.DBClusters[0].DBClusterMembers[].DBInstanceIdentifier | select(. | contains("autoscaling"))'
+)"
+
+# If there are no autoscaling nodes to delete, immediately exit
+if [[ -z $nodes_in_cluster ]]; then
+  echo "No nodes in $trimmed_cluster_id to delete"
+  exit
+fi
+
+for node in $nodes_in_cluster; do
+  # stderr is redirected to /dev/null to ensure that if there are any errors this variable is empty
+  # and no operation is done
+  instance_details_json="$(
+    aws rds describe-db-instances --db-instance-identifier "$node" 2>/dev/null || echo ""
+  )"
+
+  # If the instance is empty, then don't do anything
+  if [[ -z $instance_details_json ]]; then
+    continue
+  fi
+
+  instance_status="$(
+    echo "$instance_details_json" |
+      jq --arg trimmed_cluster_id "$trimmed_cluster_id" \
+        -r '.DBInstances[0] | select(.DBClusterIdentifier == $trimmed_cluster_id) | .DBInstanceStatus'
+  )"
+  if [[ $instance_status != "deleting" ]]; then
+    echo "Marking $node for deletion"
+    # Redirect stdout to /dev/null as the JSON response is unnecessary
+    aws rds delete-db-instance --db-instance-identifier "$node" > /dev/null
+  fi
+done
+
+echo "All autoscaling nodes in $trimmed_cluster_id marked for deletion"
+
+while true; do
+  echo "Verifying if all autoscaling nodes in $trimmed_cluster_id have been deleted..."
+
+  deleting_nodes=0
+  for node in $nodes_in_cluster; do
+    instance_details_json="$(
+      aws rds describe-db-instances --db-instance-identifier "$node" 2>/dev/null || echo ""
+    )"
+
+    # If the instance is empty, then don't do anything
+    if [[ -z $instance_details_json ]]; then
+      continue
+    fi
+
+    instance_status="$(
+      echo "$instance_details_json" |
+        jq --arg trimmed_cluster_id "$trimmed_cluster_id" \
+          -r '.DBInstances[0] | select(.DBClusterIdentifier == $trimmed_cluster_id) | .DBInstanceStatus'
+    )"
+    if [[ $instance_status == "deleting" ]]; then
+      echo "Deletion of $node is still in progress"
+      ((deleting_nodes += 1))
+    fi
+  done
+
+  if ((deleting_nodes == 0)); then
+    echo "All autoscaling nodes in $trimmed_cluster_id have been deleted"
+    exit
+  fi
+
+  echo "$deleting_nodes autoscaling node(s) have not yet been deleted. Sleeping for 5 seconds before retrying"
```

Comment:
The provisioner isn't tied to Terraform's update interval, so once it's invoked Terraform logs any output immediately and doesn't interfere with the execution of the script. In short, this logic is entirely unaffected by Terraform's internal check interval.

I could certainly increase the interval to 10 seconds, but I think the only thing that'd accomplish is to just increase the time (by a few seconds) it takes to `terraform destroy` ephemeral environments with autoscaled nodes. I'm fine with that, but it would not be necessary for the `provisioner` to work properly.